doc/user/client.md

```diff
@@ -0,0 +1,443 @@
+	err := r.client.Create(ctx, app)
+
+	...
+}
+```
+
+#### Update
+
+```Go
+// Update updates the given obj in the Kubernetes cluster. obj must be a
+// struct pointer so that obj can be updated with the content returned
+// by the API server.
+func (c Client) Update(ctx context.Context, obj runtime.Object) error
+```
+Example:
+```Go
+import (
+	"context"
+	"k8s.io/api/apps/v1"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+func (r *ReconcileApp) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	...
+	
+	dep := &v1.Deployment{}
+	err := r.client.Get(context.TODO(), request.NamespacedName, dep)
+
+	...
+
+	ctx := context.TODO()
+	dep.Spec.Selector.MatchLabels["is_running"] = "true"
+	err := r.client.Update(ctx, dep)
+
+	...
+}
+```
+
+#### Delete
+
+```Go
+// Delete deletes the given obj from Kubernetes cluster.
+func (c Client) Delete(ctx context.Context, obj runtime.Object, opts ...DeleteOptionFunc) error
+```
+A `client.DeleteOptionFunc` sets fields of `client.DeleteOptions` to configure a `Delete` call:
+```Go
+// DeleteOptionFunc is a function that mutates a DeleteOptions struct.
+type DeleteOptionFunc func(*DeleteOptions)
+
+type DeleteOptions struct {
+    // GracePeriodSeconds is the duration in seconds before the object should be
+    // deleted. Value must be non-negative integer. The value zero indicates
+    // delete immediately. If this value is nil, the default grace period for the
+    // specified type will be used.
+    GracePeriodSeconds *int64
+
+    // Preconditions must be fulfilled before a deletion is carried out. If not
+    // possible, a 409 Conflict status will be returned.
+    Preconditions *metav1.Preconditions
+
+    // PropagationPolicy determined whether and how garbage collection will be
+    // performed. Either this field or OrphanDependents may be set, but not both.
+    // The default policy is decided by the existing finalizer set in the
+    // metadata.finalizers and the resource-specific default policy.
+    // Acceptable values are: 'Orphan' - orphan the dependents; 'Background' -
+    // allow the garbage collector to delete the dependents in the background;
+    // 'Foreground' - a cascading policy that deletes all dependents in the
+    // foreground.
+    PropagationPolicy *metav1.DeletionPropagation
+
+    // Raw represents raw DeleteOptions, as passed to the API server.
+    Raw *metav1.DeleteOptions
+}
+```
+Example:
+```Go
+import (
+	"context"
+	"k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+func (r *ReconcileApp) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	...
+	
+	pod := &v1.Pod{}
+	err := r.client.Get(context.TODO(), request.NamespacedName, pod)
+
+	...
+
+	ctx := context.TODO()
+	if pod.Status.Phase == v1.PodUnknown {
+		// Delete the pod after 5 seconds.
+		err := r.client.Delete(ctx, pod, client.GracePeriodSeconds(5))
+		...
+	}
+
+	...
+}
+```
+
+#### Get
```

Comment:
Nit but can we move this section before Update and Delete. Since both of those use Get before Get is explained.